src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -235,7 +237,7 @@ await listener.AfterExecutionAsync(context)
         /// <summary>
         /// Builds a <see cref="ExecutionContext"/> instance from the provided values.
         /// </summary>
-        protected virtual ExecutionContext BuildExecutionContext(ExecutionOptions options, GraphQLDocument document, GraphQLOperationDefinition operation, Variables variables, Metrics metrics)
+        protected virtual ExecutionContext BuildExecutionContext(ExecutionOptions options, GraphQLDocument document, GraphQLOperationDefinition operation, Variables variables, Metrics metrics, IDictionary<GraphQLField, IDictionary<string, ArgumentValue>>? argumentValues, IDictionary<GraphQLField, IDictionary<string, DirectiveInfo>>? directiveValues)
```

Comment:
I would pack all arguments into struct and pass it with in modifier. Both variants binary backward incompatible but struct at least source backward compatible.

Reply:
I made a change; let's review again before adding a struct.

Reply:
OK. I plan to make a full review tonight. Was busy with other stuff.